src/credentials.rs

```diff
@@ -32,7 +31,7 @@ pub const CREDENTIALS_FILE: &str = "credentials.toml";
 #[derive(Debug, Default, Clone)]
 pub struct Credentials {
     /// A mapping from registry URIs to their corresponding tokens
-    pub registry_tokens: HashMap<RegistryUri, String>,
```

Comment:
Note: a `RegistryUri` may now be an actual `Uri`, but also an alias (resolved or not). In the rare case in this code where a URI cannot (longer) be an alias, but must be an actual URI, I have changed the type to `Uri` to enforce this guarantee.